CHANGELOG.next.asciidoc

```diff
@@ -230,6 +230,8 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - When running under Elastic-Agent Kafka output allows dynamic topic in `topic` field {pull}40415[40415]
 - The script processor has a new configuration option that only uses the cached javascript sessions and prevents the creation of new javascript sessions.
 - Update to Go 1.22.7. {pull}41018[41018]
+- Kubernetes autodiscovery now only watches metadata changes for ReplicaSets {pull}41100[41100]
+- The add_kubernetes_metadata processor now only watches metadata changes for ReplicaSets {pull}41100[41100]
```

Comment:
This is not supposed to be here.

```suggestion
```

Reply:
Where should it be? I'm making the same change to 3 different components, so I added a changelog entry for each of them.

Reply:
@swiatekm ah, my bad. I misinterpreted the intention. Normally, we only have one changelog entry per PR and it should explain the change to our users (not other developers). 

So, in this case, I'd suggest to have one entry in the `*Affecting all Beats*` section the reflects this change.

Reply:
Do you think what I have now is too detailed? I could instead do something like "Reduced memory consumption of k8s autodiscovery and the add_kubernetes_metadata processor when Deployment metadata is enabled". Does that sound better?

Reply:
Did the above.